expo/expo
# [shared-element] Update react-native-shared-element to 0.5.6

# Why

This PR updates the native react-native-shared-element code to 0.5.6.  This fixes some important issues with Images & borders on RN60+ [(see react-native-shared-element changelog)](https://github.com/IjzerenHein/react-native-shared-element/blob/master/CHANGELOG.md). Additionally, it add a slightly nicer animation for the `native-component-list` app. Lastly, it adds a shared element video to the SDK docs.

# Test Plan

- Open Expo client and run the `native-component-list`
- Go to the components tab and open the `SharedElement` item
- Clicking on the image should create a shared element transition to the next screen

Should look like this:


![sharedelement](https://user-images.githubusercontent.com/6184593/74359563-cfe3dd80-4dc3-11ea-8de3-450c7a2bdf47.gif)
